From 70d5c4bd1d411def42e90139ee3431eeedbe882a Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Wed, 20 Jun 2012 16:10:29 +0800
Subject: [PATCH 012/602] ENGR00214150 android ICS run on mx6sl board with 2D
 acceleration enable

android ICS run on mx6sl board with 2D acceleration enable

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 arm2_6sl/BoardConfig.mk | 1 +
 arm2_6sl/init.rc        | 5 ++++-
 imx6/soc/imx6sl.mk      | 6 +++---
 3 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/arm2_6sl/BoardConfig.mk b/arm2_6sl/BoardConfig.mk
index e760643..93956bb 100755
--- a/arm2_6sl/BoardConfig.mk
+++ b/arm2_6sl/BoardConfig.mk
@@ -4,6 +4,7 @@
 
 include device/fsl/imx6/soc/imx6sl.mk
 include device/fsl/imx6/arm2/BoardConfigArm2.mk
+include device/fsl-proprietary/gpu-viv/fsl-gpu.mk
 
 USE_CAMERA_STUB := true
 BOARD_HAVE_IMX_CAMERA := false
diff --git a/arm2_6sl/init.rc b/arm2_6sl/init.rc
index a676e5b..ea393ae 100755
--- a/arm2_6sl/init.rc
+++ b/arm2_6sl/init.rc
@@ -12,7 +12,7 @@ on boot
     # 3D acceleration property
     setprop debug.sf.showfps    0
     setprop debug.sf.enable_hgl 1
-    setprop debug.egl.hw	1
+    setprop debug.egl.hw	0
 
     # fsl omx graphic manager media framework property
 
@@ -50,6 +50,9 @@ on boot
     # Add support wakeup system from debug UART
     write /sys/devices/platform/imx-uart.0/tty/ttymxc0/power/wakeup enabled
 
+    # Indicate device type is tablet or SystemUI has problem without HW acceleration
+    setprop sys.devicy.type tablet
+
 service wpa_supplicant /system/bin/wpa_supplicant \
     -Dnl80211 -e/data/misc/wifi/entropy.bin
     class late_start
diff --git a/imx6/soc/imx6sl.mk b/imx6/soc/imx6sl.mk
index 8acf8e8..2f77590 100644
--- a/imx6/soc/imx6sl.mk
+++ b/imx6/soc/imx6sl.mk
@@ -4,13 +4,13 @@
 
 BOARD_SOC_TYPE := IMX6SL
 BOARD_HAVE_VPU := true
-HAVE_FSL_IMX_GPU := false
+HAVE_FSL_IMX_GPU := true
 HAVE_FSL_IMX_IPU := false
 BOARD_KERNEL_BASE := 0x80800000
 TARGET_KERNEL_DEFCONF := imx6s_android_defconfig
 HAVE_FSL_IMX_CODEC := false
-TARGET_HAVE_IMX_GRALLOC := false
-TARGET_HAVE_IMX_HWCOMPOSER = false
+TARGET_HAVE_IMX_GRALLOC := true
+TARGET_HAVE_IMX_HWCOMPOSER = true
 USE_OPENGL_RENDERER := false
 TARGET_CPU_SMP := false
 
-- 
1.8.0

